roles/ceph-osd/tasks/pool_names.yml

```diff
@@ -0,0 +1,144 @@
+---
+- assert:
+    that:
+      - ssd_pool is undefined
+      - hybrid_pool is undefined
+
+- name: gather existing cluster pool data if any
+  shell: rados lspools
+  register: existing_pools
+
+# if default pool doesn't exist
+- name: set pool names if default pool doesn't exist
+  set_fact:
+    ssd_pool: "rbd_ssd"
+    hybrid_pool: "rbd_hybrid"
+  when: '"default" not in existing_pools.stdout'
+
+# if default pool exist and ssd/hybrid pool doesn't exist
+- name: get first osd type
+  shell: ceph osd tree |sort -rn -k1 |grep -m1  -oP "(?<=host )\w+"
+  register: result_first_osd_host
+  when: ("default" in existing_pools.stdout) and
+        "rbd_ssd" not in existing_pools.stdout and
+        "rbd_hybrid" not in existing_pools.stdout
+
+- name: set pool name=default
+  set_fact:
+    ssd_pool: "default"
+    hybrid_pool: "rbd_hybrid"
+  when: result_first_osd_host is defined and
+        result_first_osd_host.stdout in groups["ceph_osds_ssd"]
+
+- name: set pool name=default
+  set_fact:
+    hybrid_pool: "default"
+    ssd_pool: "rbd_ssd"
+  when: result_first_osd_host is defined and
+        result_first_osd_host.stdout in groups["ceph_osds_hybrid"]
+
+# if default pool exists, ssd/hybrid pool exists too
+- name: set pool names
+  set_fact:
+    ssd_pool:  "rbd_ssd"
+    hybrid_pool: "default"
+  when: ("default" in existing_pools.stdout) and
+        "rbd_ssd" in existing_pools.stdout
+
+- name: set pool names
```

Comment:
Same here. I'd name the task more specific so it's easy to see which logic was true.